File: src/design_patterns/cpp/facade/Fund.cpp
#include <iostream>

#include "Fund.h"


using namespace std;


void Stock1::buy()
{
    cout << "股票1买入" << endl;
}

void Stock1::sell()
{
    cout << "股票1卖出" << endl;
}

void Stock2::buy()
{
    cout << "股票2买入" << endl;
}

void Stock2::sell()
{
    cout << "股票2卖出" << endl;
}

void Stock3::buy()
{
    cout << "股票3买入" << endl;
}

void Stock3::sell()
{
    cout << "股票3卖出" << endl;
}

void NationalDebt1::buy()
{
    cout << "国债1买入" << endl;
}

void NationalDebt1::sell()
{
    cout << "国债1卖出" << endl;
}

void Realty1::buy()
{
    cout << "房地产1买入" << endl;
}

void Realty1::sell()
{
    cout << "房地产1卖出" << endl;
}


void Fund::buyFund()
{
    m_gu1->buy();
    m_gu2->buy();
    m_gu3->buy();
    m_nd1->buy();
    m_rt1->buy();
}

void Fund::sellFund()
{
    m_gu1->sell();
    m_gu2->sell();
    m_gu3->sell();
    m_nd1->sell();
    m_rt1->sell();
}


int main()
{
    Fund * jijin = new Fund();
    jijin->buyFund();
    jijin->sellFund();
}

File: src/design_patterns/java/facade/Fund.java
package design_patterns.java.facade;

import design_patterns.java.facade.data.NationalDebt1;
import design_patterns.java.facade.data.Realty1;
import design_patterns.java.facade.data.Stock1;
import design_patterns.java.facade.data.Stock2;
import design_patterns.java.facade.data.Stock3;

public class Fund {
    private Stock1 gu1;
    private Stock2 gu2;
    private Stock3 gu3;
    private NationalDebt1 nd1;
    private Realty1 rt1;

    public Fund() {
        gu1 = new Stock1();
        gu2 = new Stock2();
        gu3 = new Stock3();
        nd1 = new NationalDebt1();
        rt1 = new Realty1();
    }

    public void buyFund() {
        gu1.buy();
        gu2.buy();
        gu3.buy();
        nd1.buy();
        rt1.buy();
    }

    public void sellFund() {
        gu1.sell();
        gu2.sell();
        gu3.sell();
        nd1.sell();
        rt1.sell();
    }
}
